packages/commons/src/middleware/userAgentMiddleware.ts

```diff
@@ -0,0 +1,38 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
```

Comment:
I don't feel very comfortable with this.

The easy way would be to just import `@aws-sdk/types` which has all the types we need. However I'm not sure we should be adding it to the bundle.

After the release I'll give it a try and see if I can do some simplified types to avoid the `@ts-nocheck` to the entire file

Reply:
I did not manage to add something like `T extends Client`, thus the shortcut. Curious to see how it can be fixed and fully agree that it's not the best way to handle it.

Reply:
I think worst case scenario we can add `// @ts-ignore` to specific lines if needed, but keep the rest of the file typed. 

Let's see, it might not be worth creating custom types to avoid issues in case the SDK changes the interfaces. What do you think?

Reply:
I have added explicit `// @ts-ignore` for two lines instead of entire file.

Do you mean to create a union type with all the clients we currently support? Something like:

`type SupportedSdkClients = LambdaClient | DynamoDBClient | S3Client;` 

It would imply that we use the `@aws-sdk/types` package, right? Or is there other way to narrow the types of the clients passed to the middleware?